tests/core/test_processor.py

```diff
@@ -76,6 +82,88 @@ def loop():
         assert json_of_latest_request(r)["message_id"] == message.message_id
 
 
+
+# This class is used for the test test_http_parsing_with_tracker in order
+# to validate that the tracker is passed into the interpreter. Also,
+# it could be used as an example of a use-case for leveraging the tracker
+# within an interpterer. In this example, the interpreter obtains a domain name
+# of the NLU endpoint from the tracker state.
+class TrackerAwareNLUHttpInterpreter(RasaNLUHttpInterpreter):
+    async def _rasa_http_parse(self, text, message_id=None, tracker=None):
```

Comment:
@ricwo I'm not quite sure what you're suggesting, could you please elaborate. What would `unittest.mock.patch()` be used to get right of `TrackerAwareNLUHttpInterpreter` I imagine? In this case and what would the test assert, instead of asserting the request's `message_id`?

Reply:
what I meant is that instead of mocking the whole class and showcasing a tracker that modifies the parse endpoint, we could instead just check that the tracker is propagated correctly to `_parse_message()`. so something along the lines of:

```python
async def mocked_parse_message(
        self, message, tracker
    ) -> None:
    # make an assertion about what should be in the tracker (e.g. the language slot)
    # and then use _parse_message() as normal
    # modify parse_data to contain information about the slot
    (...)
    parse_data["requested_language"] = tracker.get("requested_language")
    return parse_data

# set up NLU interpreter and tracker with slot
(...)
interpreter = RasaNLUInterpreter(...)

with patch.object(MessageProcessor, "_parse_message", mocked_parse_message):
    result = interpreter.parse(text, _id, tracker)
    assert result.get("requested_language") == "en"